src/tests/ecdh.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Jakub Jelen
+// See LICENSE.txt file for terms
+
+use super::tests;
+use tests::*;
+
+use serial_test::parallel;
+
+#[test]
+#[parallel]
+fn test_ecc_derive() {
+    let mut testtokn = TestToken::initialized(
+        "test_ecc_derive.sql",
+        Some("testdata/test_ecc_operations.json"),
+    );
+    let session = testtokn.get_session(true);
+
+    /* login */
+    testtokn.login();
+
+    /* private key */
+    let mut handle: CK_ULONG = CK_INVALID_HANDLE;
+    let template =
+        make_attr_template(&[], &[(CKA_UNIQUE_ID, "11".as_bytes())], &[]);
```

Comment:
This test actually reuses the ecc test data which I wanted to avoid introducing new ones. I will try to handle that later with adding some more tests.